src/osal/lv_linux.c

```diff
@@ -80,6 +85,92 @@ uint32_t lv_os_get_idle_percent(void)
     return (proc_stat.fields.idle * 100) / total;
 }
 
+uint32_t lv_os_get_self_cpu_percent(void)
+{
+    uint64_t self_current_time_ticks = 0;
+    lv_proc_stat_t stat_current_system_total_ticks;
+    lv_proc_stat_t stat_delta_system_ticks;
+
+    FILE * self = fopen(LV_UPTIME_MONITOR_SELF, "r");
+    if(!self) {
+        LV_LOG_ERROR("Failed to open " LV_UPTIME_MONITOR_SELF);
+        return LV_RESULT_INVALID;
+    }
+
+    char self_stat_buffer[LV_SELF_PROC_STAT_BUFFER_SIZE];
+
+    if(!fgets(self_stat_buffer, sizeof(self_stat_buffer), self)) {
+        fclose(self);
+        LV_LOG_ERROR("Failed to read /proc/self/stat");
+        return UINT32_MAX;
+    }
+
+    fclose(self);
+
+    /* The comm field can contain spaces and parentheses, so we find the last ')'
+     * Skip the whitespace after finding the last ')' */
+    char * p = strrchr(self_stat_buffer, ')');
+    if(!p) {
+        LV_LOG_ERROR("/proc/self/stat is missing the closing ')'");
+        return UINT32_MAX;
+    }
+    p++; /* move past the ')' */
+    while(*p && (*p == ' ' || *p == '\t'))
+        p++; /* skip whitespace after ')' */
+    if(!*p) {
+        LV_LOG_ERROR("/proc/self/stat unexpectedly ends after the closing ')'");
+        return UINT32_MAX;
+    }
+
+    uint64_t utime = 0;
+    uint64_t stime = 0;
+
+    int scanned_items = sscanf(p,
+                               "%*c "                          // state (field 3)
+                               "%*d %*d %*d %*d %*d "         // ppid, pgrp, session, tty_nr, tpgid (fields 4-8)
+                               "%*u "                         // flags (field 9)
+                               "%*" SCNu64 " %*" SCNu64 " %*" SCNu64 " %*" SCNu64 // minflt, cminflt, majflt, cmajflt (fields 10-13)
```

Comment:
```suggestion
                               "%*u %*u %*u %*u" // minflt, cminflt, majflt, cmajflt (fields 10-13)
```

This is the culprit for the error you're seeing. When ignoring something using scanf you shouldn't use a length modifier (i.e `%lu` vs `%u`). To correctly ignore these you just need to replace `%*SCNu64` with `%u`